mobile/Makefile

```diff
@@ -0,0 +1,173 @@
+## Default
+## $ make
+## sets up system for
+## $ tns run (android|ios)
+
+.DEFAULT_GOAL := apply-patches
+
+##
+## Release related targets
+##
+
+gradle-simul:
+	@( cd app/App_Resources/Android; cp app.gradle.simul app.gradle )
+
+gradle-prod:
+	@( cd app/App_Resources/Android; cp app.gradle.prod app.gradle )
+
+clean: gradle-simul
+	rm -rf node_modules platforms hooks
+
+npm-install: clean
+	npm ci
+	find app -name "*js" | xargs rm -f
+
+apply-patches: npm-install
+	patch -p0 < nodeify_temporary_patch.patch
+	cp nodeify_noxml.js node_modules/nativescript-nodeify/nodeify.js
+	rm -rf node_modules/public-encrypt/test/
+	if [ ! -d app/lib/dynacred ]; then \
+	  npm link ../dynacred; \
+	fi
+
+android-dev: apply-patches
+	tns prepare android
+
+roster-check:
+	   @if [ ! "$$PERSONHOOD3_ANDROID_PASS" ]; then echo "Please set PERSONHOOD3_ANDROID_PASS"; exit 1; fi
+	   @if grep -q -l '^export const bcDef = bdDEDIS' app/lib/byzcoin-def.ts; then \
+		 echo "Correct roster"; else \
+		 echo "\nPUT 'bcDef = bdDEDIS' IN app/lib/byzcoin-def.ts !!\n"; exit 1; fi
+
+update-version:
+	@REL=$$( grep version package.json | head -n 1 | sed -e "s/.*: .\(.*\).,/\1/" ); \
+	perl -pi -e "s/(versionName=\").*\"/\$${1}$$REL\"/" app/App_Resources/Android/src/main/AndroidManifest.xml; \
+	perl -0pi -e "s:(<key>CFBundleVersion</key>.*?<string>).*?</:\$${1}$$REL</:s" app/App_Resources/iOS/Info.plist; \
+	perl -pi -e "s/appVersion.*/appVersion = \"$$REL\";/" app/lib/user-data.ts
+
+android-compile: update-version
+	@tns build android --key-store-path c4dt-development.jks --key-store-password $$PERSONHOOD2_ANDROID_PASS \
+		--key-store-alias personhood --key-store-alias-password $$PERSONHOOD2_ANDROID_PASS --release
+	@echo "Build successful - apk is at platforms/android/app/build/outputs/apk/release/app-release.ap"
+
+android-release-copy:
+	@mkdir -p releases; \
+	REL=$$( grep versionName app/App_Resources/Android/src/main/AndroidManifest.xml | sed -e "s:.*\"\(.*\)\".*:\1:" ); \
+	cp -n platforms/android/app/build/outputs/apk/release/app-release.apk releases/personhood2.$$REL.apk
+
+android-release: roster-check apply-patches gradle-prod android-compile gradle-simul android-release-copy
+
+android-release-32-copy:
+	@mkdir -p releases; \
+	REL=$$( grep versionName app/App_Resources/Android/src/main/AndroidManifest.xml | sed -e "s:.*\"\(.*\)\".*:\1:" ); \
+	cp -n platforms/android/app/build/outputs/apk/release/app-release.apk releases/personhood2-32.$$REL.apk
+
+android-release-32: gradle-simul roster-check apply-patches android-compile android-release-32-copy
+
+android-run: roster-check gradle-simul
+	@tns run android --key-store-path c4dt-development.jks --key-store-password $$PERSONHOOD2_ANDROID_PASS \
```

Comment:
`PERSONHOOD2_ANDROID_PASS` check and let the command shine to the world :P

Reply:
check, but don't show.